File: QueuesAndStacks/StaReverseString.js
//Reverse a String using a stack in Javascript
const StaReverseString = (str) => {
    let s = [];
    let chrArray = str.split("");

    for(let i = 0; i < chrArray.length; i++) {
        s.push(chrArray[i]);
    }

    for(let i = 0; i < chrArray.length; i++) {
        chrArray[i] = s.pop();
    }
 
    return(chrArray.join(''));
}

console.log(StaReverseString("!!!ylF selgaE ylF"));

File: QueuesAndStacks/StaReverseString.java
/* Java program to reverse String using Stack */

// package QueuesAndStacks;

import java.util.*;

public class StaReverseString {
    public static void reverse(StringBuilder str) {
        int n = str.length();
        Stack<Character> obj = new Stack<>();

        for (int i = 0; i < n; i++) {
            obj.push(str.charAt(i));

        }
        // obj.forEach(System.out::println);

        for (int i = 0; i < n; i++) {
            char ch = obj.pop();
            str.setCharAt(i, ch);
        }
    }

    public static void main(String[] args) {
        StringBuilder sts = new StringBuilder("The Hawk will never die");

        reverse(sts);

        System.out.println(sts);

    }

}
